src/utils.rs

```diff
@@ -0,0 +1,968 @@
+        valid_codecs.libsvt_hevc = "".to_string();
+    }
+    if stderr.contains("libsvtav1") {
+        println!("{}",format!("libsvtav1 supported!").green());
+        valid_codecs.libsvtav1 = "libsvtav1".to_string();
+    } else {
+        println!("{}",format!("libsvtav1 not supported!").red());
+        valid_codecs.libsvtav1 = "".to_string();
+    }
+    if stderr.contains("libx265") {
+        println!("{}",format!("libx265 supported!").green());
+        valid_codecs.libx265 = "libx265".to_string();
+    } else {
+        println!("{}",format!("libx265 not supported!").red());
+        valid_codecs.libx265 = "".to_string();
+    }
+
+    let codec_support = String::from(format!("{} {} {}", valid_codecs.libsvt_hevc.to_string(), valid_codecs.libsvtav1.to_string(), valid_codecs.libx265.to_string()));
+    return codec_support;
+
+}
+
+// fn create_dirs() -> std::io::Result<()> {
+pub fn create_dirs() -> Result<(), std::io::Error> {
+    fs::create_dir_all("temp\\tmp_frames\\")?;
+    fs::create_dir_all("temp\\video_parts\\")?;
+    fs::create_dir_all("temp\\out_frames\\")?;
+    Ok(())
+}
+
+#[cfg(target_os = "linux")]
+pub fn dev_shm_exists() -> Result<(), std::io::Error> {
+    let path = "/dev/shm";
+    let b: bool = Path::new(path).is_dir();
+
+    if b == true {
+        fs::create_dir_all("/dev/shm/tmp_frames")?;
+        fs::create_dir_all("/dev/shm/out_frames")?;
+        fs::create_dir_all("/dev/shm/video_parts")?;
+        Ok(())
+    } else {
+        Err(std::io::Error::new(
+            std::io::ErrorKind::Other,
+            "dev/shm does not exist!",
+        ))
+    }
+}
+
+pub fn copy_streams_no_bin_data(
+    video_input_path: &String,
+    copy_input_path: &String,
+    output_path: &String,
+    //ffmpeg_args: &String,
+) -> std::process::Output {
+    Command::new("ffmpeg")
+        .args([
+            "-hide_banner",
+            "-v",
+            "error",
+            "-y",
+            "-i",
+            video_input_path,
+            "-i",
+            copy_input_path,
+            "-map",
+            "0:v",
+            "-map",
+            "1",
+            "-map",
+            "-1:d",
+            "-map",
+            "-1:v",
+            "-c",
+            "copy",
+            output_path
+        ])
+        .output()
+        .expect("failed to execute process")
+}
+
+pub fn copy_streams(
+    video_input_path: &String,
+    copy_input_path: &String,
+    output_path: &String,
+) -> std::process::Output {
+    Command::new("ffmpeg")
+        .args([
+            "-hide_banner",
+            "-v",
+            "error",
+            "-y",
+            "-i",
+            video_input_path,
+            "-i",
+            copy_input_path,
+            "-map",
+            "0:v",
+            "-map",
+            "1",
+            "-map",
+            "-1:v",
+            "-c",
+            "copy",
+            output_path
+        ])
+        .output()
+        .expect("failed to execute process")
+}
```

Comment:
This can be simplified and combined with `fn merge_video_parts()`.
```suggestion
pub fn concatenate_segments(&self) {
        let mut f_content = String::from("file 'video_parts\\0.mp4'");
        for segment_index in 1..self.segment_count {
            let video_part_path = format!("video_parts\\{}.mp4", segment_index);
            f_content = format!("{}\nfile '{}'", f_content, video_part_path);
        }
        fs::write("temp\\parts.txt", f_content).unwrap();

        Command::new("ffmpeg")
            .args([
                "-f",
                "concat",
                "-safe",
                "0",
                "-i",
                "temp\\parts.txt",
                "-i",
                &self.path,
                "-map",
                "0:v",
                "-map",
                "1:a?",
                "-map",
                "1:s?",
                "-map_chapters",
                "1",
                "-c",
                "copy",
                &self.output_path,
            ])
            .output()
            .unwrap();
        fs::remove_file("temp\\parts.txt").unwrap();
    }
```